Compiling with kokkos and nvcc_wrapper fails

I've been working on getting compiling a Lammps version with Kokkos (built with CUDA support) and ML-Pace.

I have an issue when I try to compile a specific file:
`ML-PACE/ace/ace_clebsch_gordan.cpp`

specifically due to a problem with the include file `wigner/wigner_3nj.hpp`. I realize that wigner seems to be an external repository, but no activity has happened there since 2017 so I suppose you're the best option to figure this out.

I have an error that seems to concern line 215 of that hpp file. Here's the output of `make VERBOSE=1` for the ace_clebsch_gordan.cpp file:
```
/path/to/kokkos/bin/nvcc_wrapper -DEXTRA_C_PROJECTIONS -I/path/to/lammps/build/lammps-user-pace-v.2023.01.3/ML-PACE -I/path/to/lammps/build/lammps-user-pace-v.2023.01.3/yaml-cpp/include -I/path/to/lammps/build/lammps-user-pace-v.2023.01.3 -I/path/to/lammps/build/lammps-user-pace-v.2023.01.3/wigner-cpp/include -O3 -DNDEBUG -fPIC -std=gnu++14 -MD -MT build-pace/CMakeFiles/pace.dir/ML-PACE/ace/ace_clebsch_gordan.cpp.o -MF CMakeFiles/pace.dir/ML-PACE/ace/ace_clebsch_gordan.cpp.o.d -o CMakeFiles/pace.dir/ML-PACE/ace/ace_clebsch_gordan.cpp.o -c /path/to/lammps/build/lammps-user-pace-v.2023.01.3/ML-PACE/ace/ace_clebsch_gordan.cpp
/path/to/lammps/build/lammps-user-pace-v.2023.01.3/wigner-cpp/include/wigner/wigner_3nj.hpp: In function 'void wigner::detail::rec_luscombe_luban(ArgT, ArgT, Iterator, FuncXYZ)':
/path/to/lammps/build/lammps-user-pace-v.2023.01.3/wigner-cpp/include/wigner/wigner_3nj.hpp:215:13: error: invalid use of 'auto'
  215 |     auto i1         = decltype(siz)();
      |             ^~~~
```

This error only shows up on GPU builds. A Kokkos build with no GPU support (so, using mpicxx as `CMAKE_CXX_COMPILER` instead of the nvcc_wrapper) compiles fine.

Here's what I tried already with Kokkos-3.7.1:
- a Kokkos built with gcc-8.4.0 with CUDA 10.2.89 (the standard on our cluster);
- a Kokkos built with gcc-6.5.0 with CUDA 10.2.89 (a gcc more along the lines of what was available when wigner came out, in case the issue was coming from recent compilers being more strict with standards)
- a Kokkos built with with gcc-6.5.0 with CUDA 9.2.88 (the oldest CUDA for which Kokkos is known to work, following the logic described above)
- a Kokkos built with gcc-11.3.0 with CUDA 11.7.0 (in case my logic was completely wrong and the opposite was actually true).
- I also tried Kokkos-3.6.1 built gcc-8.4.0 and CUDA 10.2.89 with the same results.

These were built using Spack, so the builds are consistent. An initial Kokkos build (by hand, so I could validate the compilation options) gave the same results.
For the record, here's the minimal cmake command I've been trying:
```
cmake ../cmake/ -DBUILD_MPI=OFF -DBUILD_OMP=ON \
  -DPKG_ML-PACE=ON -DPKG_KOKKOS=ON \
  -DKokkos_ARCH_HOSTARCH=SKX -DKokkos_ARCH_GPUARCH=VOLTA70 \
  -DKokkos_ENABLE_CUDA=yes -DKokkos_ENABLE_OPENMP=yes \
  -DCMAKE_CXX_COMPILER=$(which nvcc_wrapper)
```

Let me know if there's any other piece of information you may need.

--- a/wigner-cpp/include/wigner/wigner_3nj.hpp
+++ b/wigner-cpp/include/wigner/wigner_3nj.hpp
@@ -212,7 +212,7 @@ void rec_luscombe_luban(ArgT nmin_, ArgT nmax_, Iterator dest, const FuncXYZ fn)
     //
     // For the convenience, we set f'[n1] = 1.
     //
-    auto i1         = decltype(siz)();
+    auto i1         = 0;
     auto n1         = nmin;
     const auto ymin = fn.Y(nmin);
     if (abs(ymin) < tiny)

